components/resolved_ts/src/advance.rs

```diff
@@ -435,10 +425,16 @@ impl LeadershipResolver {
             match res {
                 Ok((to_store, resp)) => {
                     for region_id in resp.regions {
-                        resp_map
-                            .entry(region_id)
-                            .or_insert_with(|| Vec::with_capacity(store_count))
-                            .push(to_store);
+                        if let Some(prog) = progresses.get_mut(&region_id) {
+                            if prog.resolved {
```

Comment:
Should the quorum also be checked when the `res` is `Err(...)` at https://github.com/tikv/tikv/pull/16036/files#diff-f86dc21a7845465caa70ed3ebe51652da2bb799b200680dd5f210701bb0c96d9R441?
For now it seems the `regiond` ids for each store RPC is not recorded yet.

Reply:
This PR is just the cherry-pick of #15567. It's only add a short circuit when all regions are resolved. I think we can check `res` is `Err` and keep track a list of regions that we are sure they cannot be resolved, then break on `valid_regions.len() + invalid_regions.len() >= progresses.len()`. How about to do this optimization in next PR?